sharktank_models/punet/sharded_toy_resnet_block/test_sharded_toy_resnet_block.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2025 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+
+from typing import Optional, Union
+import pytest
+import pathlib
+from os import PathLike
+import os
+import numpy as np
+import ml_dtypes
+
+import iree.runtime
+import iree.compiler
+
+THIS_DIR = pathlib.Path(__file__).parent
+
+shard_count = 2
+
+
+def load_tensors_from_irpa(path: PathLike) -> list[np.ndarray]:
+    index = iree.runtime.ParameterIndex()
+    index.load(str(path))
+    index_dict = dict(index.items())
+    index_dict = {k: v for k, v in index_dict.items() if k.isdigit()}
+    return [
+        iree.runtime.parameter_index_entry_as_numpy_ndarray(index_dict[f"{i}"])
+        for i in range(len(index_dict.items()))
+    ]
+
+
+@pytest.fixture(
+    params=[
+        pytest.param("local-task", marks=pytest.mark.target_cpu),
+        pytest.param("hip", marks=pytest.mark.target_hip),
+    ]
+)
+def device_id(request: pytest.FixtureRequest) -> str:
+    return request.param
+
+
+@pytest.fixture(params=["f32"])
+def model_variant(request: pytest.FixtureRequest) -> str:
+    return request.param
+
+
+def get_devices(driver: str) -> list[iree.runtime.HalDevice]:
+    if driver == "local-task":
+        return [iree.runtime.get_device(driver)] * shard_count
+
+    # Get as many devices as are available if they are less then required,
+    # repeat them to get the required count.
+    hal_driver = iree.runtime.get_driver(driver)
+    available_device_count = len(hal_driver.query_available_devices())
+    device_ids = [f"{driver}://{i%available_device_count}" for i in range(shard_count)]
+    return [iree.runtime.get_device(d) for d in device_ids]
+
+
+mlir_path = {
+    "f32": THIS_DIR / "assets/model.mlir",
+}
+
+parameters_path = {
+    "f32": [THIS_DIR / f"assets/model.rank{i}.irpa" for i in range(shard_count)],
+}
+
+input_args_path = {
+    "f32": THIS_DIR / "assets/input_args.irpa",
+}
+
+expected_results_path = {
+    "f32": THIS_DIR / "assets/expected_results.irpa",
+}
+
+absolute_tolerance = {
+    "f32": 5e-5,
+}
+
+
+def compiler_args(device_id: str) -> list[str]:
+    res = []
+    if device_id == "local-task":
+        iree_hal_target_device = "llvm-cpu"
```

Comment:
Please incorporate the changes from https://github.com/nod-ai/shark-ai/pull/1122 here

Reply:
Like this: https://github.com/iree-org/iree-test-suites/pull/90